src/plugins/nestedRows/data/dataManager.js

```diff
@@ -96,6 +96,10 @@ class DataManager {
    * @private
    */
   rewriteCache() {
+    if (!this.data?.length) {
```

Comment:
```suggestion
    if (Array.isArray(data) === false || data.length === 0) {
```

In my opinion, it would me more clear, but it's up to you. But, probably it won't be needed after another change.